File: Python ejs/UD03/UD03A033.py
a=int(input("Introduzca un coeficiente: "))
b=int(input("Introduzca otro coeficiente: "))

if a==0 and b==0:
    print("LA ECUACIÓN TIENE INFINITAS SOLUCIONES")
elif a==0:
    print("LA ECUACIÓN NO TIENE SOLUCIÓN")
else:
    print("LA ECUACIÓN TIENE UNA SOLUCIÓN Y ES "+ (-b/a))

File: Java ejs/UD3/A03/UD03A033.java
import java.util.Scanner;
public class UD03A033{
	public static void main(String[] args){
		int a,b;
        Scanner teclado=new Scanner(System.in);
        System.out.println("Introduzca un coeficiente: ");
        a=teclado.nextInt();
        System.out.println("Introduzca un coeficiente: ");
        b=teclado.nextInt();
        if (a==0 && b==0)
        {
            System.out.println("LA ECUACIÓN TIENE INFINITAS SOLUCIONES");
        }else{
            if (a==0)
            {
                System.out.println("LA ECUACIÓN NO TIENE SOLUCIÓN");
            }else{
                System.out.println("LA ECUACIÓN TIENE UNA SOLUCIÓN Y ES "+ (-b/a));
            }
        }
	}
}